src/Pdp/PublicSuffixListManager.php

```diff
@@ -63,10 +70,22 @@ public function __construct($cacheDir = null)
     public function refreshPublicSuffixList()
```

Comment:
I'd like to refactor this in such a way that it parses the PSL in a single pass. As it is now it reads the PSL once for each pass. I think we could get a nice performance boost as well as a significant reduction in complexity. Would you be willing to do that?